src/java/src/main/java/ai/onnxruntime/genai/Tensor.java

```diff
@@ -62,11 +62,40 @@ public Tensor(ByteBuffer data, long[] shape, ElementType elementType) throws Gen
 
     this.elementType = elementType;
     this.shape = shape;
-    this.dataBuffer = data;  // save a reference so the owning buffer will stay around.
+    this.dataBuffer = data; // save a reference so the owning buffer will stay around.
 
     nativeHandle = createTensor(data, shape, elementType.ordinal());
   }
 
+  /**
+   * Construct a Tensor from native handle.
+   *
+   * @param handle The native tensor handle.
+   */
+  public Tensor(long handle) {
```

Comment:
How do you get a nativeHandle for a Tensor outside of this class? Naively I would have expected the handle to remain an internal implementation detail and at the Java level the user would only deal with the Tensor class.

Reply:
You're right. This should be internal.